src/blob.rs

```diff
@@ -1,86 +1,258 @@
-use std::path::Path;
+use futures::{
+    future::Future,
+    task::{Poll, Waker},
+};
+use std::{
+    collections::BTreeMap,
+    fs, io,
+    os::unix::fs::FileExt,
+    path::{Path, PathBuf},
+    pin::Pin,
+};
 
-use crate::record::Record;
+use crate::record::{self, Record};
+
+const BLOB_MAGIC_BYTE: u64 = 0xdeaf_abcd;
+
+type Result<T> = std::result::Result<T, Error>;
 
 /// A `Blob` struct for performing of database,
-#[derive(Debug, Default)]
-pub struct Blob<T> {
+#[derive(Debug)]
+pub struct Blob {
+    index: BTreeMap<Vec<u8>, record::Header>,
```

Comment:
If I understand correctly, this is an in-memory index. BTree is good for disk indexing. For memory there are better data structures: hash map, binary tree.

Reply:
Why are they better?
[HashMap vs BTreeMap](https://www.reddit.com/r/rust/comments/7rgowj/hashmap_vs_btreemap/)
Rust collections [usecases](https://doc.rust-lang.org/std/collections/index.html)

Reply:
this is temporary - don't pick on this - we will implement in future something suitable